### ISSUE TITLE
Assassin's Creed Brotherhood shadows flickering issue

### ISSUE THREAD

**Issue by CTRL-ALT-DEL-92** [iXit/wine-nine-standalone#168]
Dear G9 Devs,

I'm currently running ACB via G9 with outstanding performances (my gpu is AMD Ellesmere [Radeon RX 470/480/570/570X/580/580X/590] vendor: Sapphire Nitro+), the only  hideous issue is with some very eyes hurting shadows flickering on the protagonist cape and costume. The solution I found to circumvent this problem was changing the protagonist outfit to a totally black version of his costume, with the white one the flickering is almost eyes damaging. 
![Istantanea_2024-06-30_15-14-43](https://github.com/iXit/wine-nine-standalone/assets/78966095/387a8b5a-6740-434e-8c8a-42ebd17ad461)
I've successfully made an apitrace to help you find a fix. 
https://drive.google.com/file/d/1WdU6N_qyIQ4yTkvuOfAKh8nfluBIVekk/view?usp=sharing

Thanks in advance for your help

**Comment by axeldavy** [comment]
Tried the trace and could reproduce. This looks like some rounding issues, possibly related to the depth buffer.
If you play with graphic settings in the game (with and without antialiasing, etc) is there any impact on the issue ?

Thanks

**Comment by dungeon007** [comment]
And i cant reproduce this on CIK APU... so this sounds like hw specific, maybe Polaris issue.
So play with AMD_DEBUG flags, maybe it is nodcc or something of that sort... wild random guess because CIK do not have DCC.

**Comment by CTRL-ALT-DEL-92** [comment]
I've finally enabled the AMD_DEBUG=nodcc flag but with no avail... The game isn't affected, just changing the character costume colour affects the problem.

**Comment by dungeon007** [comment]
Bug is fine, but is not actually hw specific... it looks like REGRESSION since commit 55d81214c9d800c0667337808a82143ebab17c1a

**radeonsi: replace gl_FrontFacing with a constant if one side is always culled**

**Comment by CTRL-ALT-DEL-92** [comment]
> Bug is fine, but probably not actually hw specific... it looks like REGRESSION bug since last year commit 55d81214c9d800c0667337808a82143ebab17c1a
> 
> **radeonsi: replace gl_FrontFacing with a constant if one side is always culled**

Should  the mesa developers be informed of this?

**Comment by dungeon007** [comment]
Yeah, i already opened a bug there https://gitlab.freedesktop.org/mesa/mesa/-/issues/11484
